frontend/src/services/apiFriends.ts

```diff
@@ -0,0 +1,20 @@
+import axios from 'axios';
+
+axios.defaults.baseURL = 'http://127.0.0.1:8000/api/user/';
+
+export const getFriends = async () => {
+  const friendsAPIurl = document.location.pathname.match(/user\/[0-9a-f]{24}\/friends/);
+  const userId = friendsAPIurl ? friendsAPIurl[0].split('/')[1] : null;
```

Comment:
the id should come from component and not parsed like this in service
take a look here  https://reactrouter.com/web/api/Hooks/useparams